File: commonutils/src/main/java/net/maxsmr/commonutils/data/gson/exclusion/FieldsAnnotationExclusionStrategy.java
package net.maxsmr.commonutils.data.gson.exclusion;

import android.support.annotation.Nullable;

import com.google.gson.ExclusionStrategy;
import com.google.gson.FieldAttributes;

/** @see net.maxsmr.commonutils.data.gson.exclusion.FieldExclude */
class FieldsAnnotationExclusionStrategy implements ExclusionStrategy {

    @Override
    public boolean shouldSkipField(@Nullable FieldAttributes f) {
        if (f != null) {
            return f.getAnnotation(FieldExclude.class) != null;
        }
        return false;
    }

    @Override
    public boolean shouldSkipClass(Class<?> clazz) {
        return false;
    }
}

File: commonutils-jre/src/main/kotlin/net/maxsmr/commonutils/model/gson/exclusion/FieldsAnnotationExclusionStrategy.kt
package net.maxsmr.commonutils.model.gson.exclusion

import com.google.gson.ExclusionStrategy
import com.google.gson.FieldAttributes

class FieldsAnnotationExclusionStrategy : ExclusionStrategy {

    override fun shouldSkipField(f: FieldAttributes?): Boolean {
        return if (f != null) {
            f.getAnnotation(FieldExclude::class.java) != null
        } else {
            false
        }
    }

    override fun shouldSkipClass(clazz: Class<*>?): Boolean = false
}